src/main.ts

```diff
@@ -50,11 +50,15 @@ async function run(): Promise<undefined> {
       header
     )
 
-    if (!previous) {
+    if (!previous && !deleteOldComment) {
       await createComment(octokit, repo, pullRequestNumber, body, header)
       return
     }
 
+    if (!previous) {
+      return
+    }
+
```

Comment:
@marocchino Do you mean this? (Same result, fewer lines)
```suggestion
    if (!previous) {
      if (!deleteOldComment) {
        await createComment(octokit, repo, pullRequestNumber, body, header)
      }

      return
    }

```

Reply:
@awkaiser-tr umm.. no.
I want halt delete or hide logic before previous check.